### DISCUSSION TITLE
Formatter for Kind

### DISCUSSION THREAD

**Discussion by dvberkel** [uwu-tech/Kind#282] [Ideas]
There are other languages that have a de-facto way of formatting their code, and implement the standard in a tool that can be applied to your code base. The motivation is that it eliminates discussions on formatting, frees the developer from fidgeting with formatting and let them focus on more interesting ideas.

I personally like to have a tool that formats my code. But I know others might frown upon giving up their freedom when it comes to laying out code. I started this discussion to gauge the waters and see if there is an interest in developing a formatter for Kind. And if there is, how it could/should be implemented.

**Comment by huynhtrankhanh** [comment]
I support this feature request. This is a feature that is somewhat easy to implement but it would immensely enhance the user experience. :tada: :+1: :100:

If you don't know what a formatter is, well, [here is one](https://prettier.io/).

**Comment by dvberkel** [comment]
From the [telegram chat](https://t.me/formality_lang/17766):

> @rigille  the easiest way to do that would be to parse the string with `Kind.Parser.file` and then format it with a `Kind.Term -> String function`
